### ISSUE TITLE
Locale parameter is not being stored reliably in URL alias

### ISSUE THREAD

**Issue by nimajneb** [strapi-community/strapi-plugin-url-alias#6]
### Describe the bug

Nice plugin! I'm currently trying to work around Strapi's limitation of not being able to have a UID field that checks for uniqueness only across entries with the same locale. At the same time I need a way of fetching an entry by its slug. So this plugin looks very promising. :)

I've run into a bit of an issue however, where I have a URL pattern configured to contain the entry locale, but it's not consistently stored there.

### Steps to reproduce the behavior

1. Create a URL pattern with the entry locale in it, e.g. `/products/[locale]/[title]`.
2. Create an entry in the default locale (de-DE in my case), save and publish it. The locale is correctly stored in the URL alias.
3. Change the entry and save again. The locale disappears from the URL alias. (I checked the database, it's really gone.)
4. Unpublish the entry. The locale is again stored to the URL alias.
5. Repeat step 3. The locale is gone again.
6. Publish the entry. The locale is there again.

### Expected behavior

The locale parameter should stay in the URL regardless of the entry's state changes.

### System

- Node.js version: 16.16.0 LTS
- NPM version: 8.11.0
- Strapi version: 4.3.2
- Plugin version: 1.0.0-alpha.4
- Database: SQLite
- Operating system: macOS Monterey v12.2

**Comment by boazpoolman** [comment]
Hey @nimajneb 

Thanks for reporting this issue! I'll look in to it.

**Comment by boazpoolman** [comment]
@nimajneb I've fixed this issue in the `master` branch.
Could you test it out?

```
yarn add strapi-community/strapi-plugin-url-alias
npm install strapi-community/strapi-plugin-url-alias
```

**Comment by nimajneb** [comment]
@boazpoolman Oh great, thanks for taking care of this! :)

The locale param now is stored as expected, but now another strange thing happens - the URL alias seems to always be one save "behind":

Steps to reproduce:

1. Update plugin (`yarn add ... && yarn build && yarn develop`)
2. URL pattern: `/products/[locale]/[title]`
3. Create new entry, title `a` -> URL alias is correctly stored as `/products/en/a`
4. Update title to `b`-> URL alias is not updated (but keeps the locale)
5. Update title to `c` -> URL alias is updated to `products/en/b`

I didn't do a fresh install but only updated the plugin and rebuilt the backend - that should make no difference, right?

**Comment by boazpoolman** [comment]
Ah good eye! I missed this myself. It's a bug I introduced trying to fix this.
Thanks. I'll take a look at this.

**Comment by boazpoolman** [comment]
This was released with version `1.0.0-alpha.5` of the plugin.
If you're still experiencing the issue beyond this version feel free to re-open the GH issue or create a new one.
